hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/InputPathHandler.java

```diff
@@ -53,25 +57,27 @@
   public static final Logger LOG = LogManager.getLogger(InputPathHandler.class);
 
   private final Configuration conf;
-  // tablename to metadata mapping for all Hoodie tables(both incremental & snapshot)
+  // tableName to metadata mapping for all Hoodie tables(both incremental & snapshot)
   private final Map<String, HoodieTableMetaClient> tableMetaClientMap;
   private final Map<HoodieTableMetaClient, List<Path>> groupedIncrementalPaths;
   private final List<Path> snapshotPaths;
   private final List<Path> nonHoodieInputPaths;
+  private boolean isIncrementalUseDatabase;
 
-  public InputPathHandler(Configuration conf, Path[] inputPaths, List<String> incrementalTables) throws IOException {
+  public InputPathHandler(Configuration conf, Path[] inputPaths, List<String> incrementalTables, JobConf job) throws IOException {
     this.conf = conf;
     tableMetaClientMap = new HashMap<>();
     snapshotPaths = new ArrayList<>();
     nonHoodieInputPaths = new ArrayList<>();
     groupedIncrementalPaths = new HashMap<>();
+    this.isIncrementalUseDatabase = HoodieHiveUtils.isIncrementalUseDatabase(Job.getInstance(job));
```

Comment:
`JobConf` is passed in just to compute a boolean config? can we somehow make the config extracted from `Configuration conf` ? always prefer to have less args

Reply:
Yes, We can. Thank you for your reminder. That's a good idea